main.nf

```diff
@@ -36,203 +35,243 @@ if (!svvcf2bed.exists()) {
+    "https://github.com/lh3/varcmp/raw/master/scripts/LCR-hs37d5.bed.gz"
+]
+
+Channel.from( 0..<mask_urls.size() ).map { [it, mask_urls[it]] }.set { mask_urls_channel }
+
+process download_masks {
+    input:
+        set val(index), val(mask_url) from mask_urls_channel
+    output:
+        file 'mask_*.bed.gz' into masks
+
+    // We only need one core for this part
+    clusterOptions = {
+        "-A $params.project -p $params.runspecs.core $params.runspecs.extra"
     }
 
-    process run_intersections {
-        input:
-            file 'manta.bed' from manta_bed
-            file 'fermi.bed' from fermi_bed
-            file 'mask_ceph.bed' from mask_ceph
-            file 'mask_lcr.bed' from mask_lcr
-        output:
-            file 'manta_masked*.bed'
-            file 'fermi_masked*.bed'
-            file 'combined*.bed'
+    """
+    wget -O mask_${index}.bed.gz $mask_url
+    """
+}
 
-        publishDir 'results'
 
-        // We only need one core for this part
-        clusterOptions = {
-            "-A $params.project -p core"
-        }
+// Collect both bed files and combine them with the mask files
+beds = manta_bed.mix( fermi_bed )
+beds.spread( masks.buffer(size: 2) ).set { mask_input }
 
-        module 'bioinfo-tools'
-        module "$params.modules.bedtools"
+process mask_beds {
+    input:
+        set file(bedfile), file(mask1), file(mask2) from mask_input
+    output:
+        file '*_masked.bed' into masked_beds
+        file '*_masked_*.bed'
 
-        """
-        cat manta.bed \
-            | bedtools intersect -v -a stdin -b mask_lcr.bed -f 0.25 \
-            | bedtools intersect -v -a stdin -b mask_ceph.bed -f 0.25 > manta_masked.bed
-        cat fermi.bed \
-            | bedtools intersect -v -a stdin -b mask_lcr.bed -f 0.25 \
-            | bedtools intersect -v -a stdin -b mask_ceph.bed -f 0.25 > fermi_masked.bed
-
-        ## In case grep doesn't find anything we want to continue
-        set +e
-
-        ## Create filtered bed files
-        for FILE in manta_masked fermi_masked; do
-            for WORD in DEL INS DUP; do
-                grep -w \$WORD \$FILE.bed > \$FILE.\$WORD.bed
-            done
-        done
-
-        ## But now we want to abort in case of errors
-        set -e
-
-        ## Create intersected bed file
-        for WORD in DEL INS DUP; do
-            intersectBed -a fermi_masked.\$WORD.bed -b manta_masked.DEL.bed -f 0.5 -r | sort -k1,1V -k2,2n > combined_masked.SV.\$WORD.bed
-        done
-        """
+    publishDir 'results'
+
+    clusterOptions = {
+        "-A $params.project -p $params.runspecs.core $params.runspecs.extra"
     }
+
+    module 'bioinfo-tools'
+    module "$params.modules.bedtools"
+
+    """
+    BNAME=\$( echo $bedfile | cut -d. -f1 )
+    MASK_FILE=\${BNAME}_masked.bed
+    cat $bedfile \
+        | bedtools intersect -v -a stdin -b $mask1 -f 0.25 \
+        | bedtools intersect -v -a stdin -b $mask2 -f 0.25 > \$MASK_FILE
+
+
+    set +e
```

Comment:
Maybe good to comment? (I had personally forgot what this does ... vague memory of some bash settings ..)